float32 earth_x
float32 earth_y
float32 earth_z
